examples/fabric/reinforcement_learning/train_fabric.py

```diff
@@ -146,7 +146,9 @@ def main(args: argparse.Namespace):
             # Single environment step
             next_obs, reward, done, truncated, info = envs.step(action.cpu().numpy())
             done = torch.logical_or(torch.tensor(done), torch.tensor(truncated))
-            rewards[step] = torch.tensor(reward, device=device).view(-1)
+            rewards[step] = torch.tensor(
+                reward, device=device, dtype=torch.float32 if device.type == "mps" else None
+            ).view(-1)
```

Comment:
Setting `dtype=torch.float32` unconditionally should also be fine I think.

Reply:
Thank you :) 
I applied your feedback!